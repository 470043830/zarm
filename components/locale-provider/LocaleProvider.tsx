import React, { PureComponent } from 'react';
import createContext, { Context } from 'create-react-context';
import { LocaleProviderProps, Locale } from './PropsType';

const defaultLocale = {} as any as Locale;
export const LocaleContext: Context<Locale> = createContext(defaultLocale);

export default class LocaleProvider extends PureComponent<LocaleProviderProps, {}> {
  static defaultProps = {
    locale: {},
  };

  render() {
    const { children, locale } = this.props;
    return (
      <LocaleContext.Provider value={locale}>
        {React.Children.only(children)}
      </LocaleContext.Provider>
    );
  }
}
