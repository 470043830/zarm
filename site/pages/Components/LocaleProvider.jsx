import Markdown from '@site/components/Markdown';

export default class Page extends Markdown {
  document = () => {
    return {
      document: require('@/components/locale-provider/demo.md'),
      className: 'locale-provider-page',
    };
  }
}
