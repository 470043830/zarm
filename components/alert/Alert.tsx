import React, { PureComponent } from 'react';
import ReactDOM from 'react-dom';
import classnames from 'classnames';
import PropsType from './PropsType';
import Modal from '../modal';
import alertLocale from './locale/zh_CN';
import { getRunTimeLocale } from '../locale-provider/LocaleProvider';

export interface AlertProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Alert extends PureComponent<AlertProps, {}> {
  static defaultProps = {
    prefixCls: 'za-alert',
    animationType: 'zoom',
    message: '',
    locale: alertLocale,
  };

  // private static alertContainer;

  // static show = (props: AlertProps) => {
  //   const { defaultProps } = Alert;
  //   let _props: any;

  //   if (typeof props === 'object') {
  //     _props = { ...defaultProps, ...props };
  //   } else {
  //     _props = { ...defaultProps };
  //   }

  //   const runTimeLocale = getRunTimeLocale();
  //   if (runTimeLocale && runTimeLocale.Alert) {
  //     _props = { ..._props, locale: runTimeLocale.Alert };
  //   }
  //   Alert.alertContainer = document.createElement('div');
  //   document.body.appendChild(Alert.alertContainer);
  //   return new Promise((resolve) => {
  //     ReactDOM.render(
  //       <Alert
  //         {..._props}
  //         visible
  //         onCancel={() => {
  //           if (typeof _props.onCancel === 'function') {
  //             _props.onCancel();
  //           }
  //           resolve(false);
  //         }}
  //       />,
  //       Alert.alertContainer,
  //     );
  //   });
  // };

  // static hide = () => {
  //   if (Alert._hide) {
  //     Alert._hide();
  //   }
  // };

  // static _hide: () => void;

  state = {
    visible: this.props.visible,
  };

  // componentDidMount() {
  //   Alert._hide = this._hide;
  // }

  componentWillReceiveProps(nextProps) {
    const { visible } = this.props;

    if (nextProps.visible !== visible) {
      if (nextProps.visible === true) {
        this.setState({
          visible: true,
        });
      } else {
        this._hide();
      }
    }
  }

  _hide = () => {
    this.setState({
      visible: false,
    });
  };

  afterClose = () => {
    const { afterClose } = this.props;
    // if (Alert.alertContainer) {
    //   document.body.removeChild(Alert.alertContainer);
    //   Alert.alertContainer = null;
    // }

    if (typeof afterClose === 'function') {
      afterClose();
    }
  };

  render() {
    const { prefixCls, className, message, cancelText, onCancel, locale, ...others } = this.props;
    const { visible } = this.state;
    const cls = classnames(prefixCls, className);

    return (
      <Modal
        className={cls}
        {...others}
        visible={visible}
        afterClose={this.afterClose}
        footer={<div className={`${prefixCls}__button`} onClick={onCancel}>{cancelText || locale!.cancelText}</div>}
      >
        {message}
      </Modal>
    );
  }
}
