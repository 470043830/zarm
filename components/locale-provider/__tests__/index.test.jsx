import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import LocaleProvider from '../index';
import Alert from '../../alert';
import localeCN from '../locale/zh_CN';

describe('LocaleProvider', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <LocaleProvider locale={localeCN}>
        <Alert />
      </LocaleProvider>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
