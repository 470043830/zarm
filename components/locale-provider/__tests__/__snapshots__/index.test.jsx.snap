// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleProvider renders correctly 1`] = `null`;
